### PR TITLE
 ref: Accept client-side wayland connection, global

### PR DESCRIPTION
**Pull request by Decodetalkers**

### CODE REVIEW

File: libwaysip/src/lib.rs [hunk 1]
```diff
@@ -35,12 +35,35 @@ fn get_cursor_buffer(connection: &Connection, shm: &WlShm) -> Option<CursorImage
...
+    let (connection, globals) = match waysip_connection {
+        Some(WaysipConnection {
+            connection,
+            globals,
+        }) => (connection.clone(), globals),
+        None => {
+            let connection =
+                Connection::connect_to_env().map_err(|e| WaySipError::InitFailed(e.to_string()))?;
+            let (globals, _) = registry_queue_init::<state::WaysipState>(&connection)
+                .map_err(|e| WaySipError::InitFailed(e.to_string()))?; // We just need the
```
Comment (Shinyzenith) [inline libwaysip/src/lib.rs:57]: Comment is outdated. BaseState no longer exists. We can just delete this comment.

**Review by Shinyzenith — CHANGES_REQUESTED** [review]

**Comment by Decodetalkers** [comment]
@Shinyzenith may I merge this one? will that be ok?

**Comment by Decodetalkers** [comment]
ok

File: libwaysip/src/lib.rs (hunk 1 above)
Comment (Decodetalkers) [reply libwaysip/src/lib.rs]: done

**Comment by Shinyzenith** [comment]
Thank you for the help @Decodetalkers ❤️ I'll keep making a few more improvements and get back to you!

**Pull request merged**